### ISSUE TITLE
Use local persistent volumes for embedded journal, single master local journal, off-heap metastore and tiered storage

### ISSUE THREAD

**Issue by madanadit** [Alluxio/alluxio#10253]
**Is your feature request related to a problem? Please describe.**
Local persistent volumes is the preferred way for storage to persist across restarts of a pod (ref: https://kubernetes.io/docs/concepts/storage/volumes/#local). We should use persistent volumes for the following:
1. embedded journal volume for master pods
2. single master journal volume for master pod
3. off-heap metastore volume for master pods
4. tiered storage for worker pods

**Describe the solution you'd like**
Helm charts should be extended to support local persistent volumes w/ resource limits (e.g., size of journal / tiered storage). Documentation w/ and w/o helm should be added as part of this ask. Note: hostPath volumes cannot be provisioned with resource limits and require root user and are hence not desired for persistent local storage.

1. embedded journal volume for master pods: restarts of the master pod w/o persistent volumes will lead to loosing metadata. 
- The helm template `statefulset.yaml` should use persistentVolumeClaims instead of `emptyDir` volumes. 
- `journal-pvc.yaml` should create `masterCount` claim resources
- `journal-pv.yaml` should be changed to use local persistent volume instead of `hostPath` by default

2. single master journal volume for master pod
- As part of pv for embedded journal we may choose to rename the claim from `alluxio-pv-claim` to `alluxio-pv-claim-<i>` where `i=0` for a single master. 

3. off-heap metastore volume for master pods
- Switch from `emptyDir` volume to localPV with limits

4. tiered storage for worker pods
- Switch from unbounded `hostPath` volume to localPV with limits

**Describe alternatives you've considered**
Advanced end-user customizes the specs on their own

**Urgency**
It would be painful w/o extensive knowledge of Alluxio to roll out raw specifications for these features to be production ready. This deployment will not be possible w/ helm (w/o this feature).

**Additional context**
N/A

**Comment by madanadit** [comment]
@cheyang FYI

**Comment by jiacheliu3** [comment]
Ok

**Comment by jiacheliu3** [comment]
1) and 2) are resolved by https://github.com/Alluxio/alluxio/pull/10686

**Comment by jiacheliu3** [comment]
After some research, 4(worker tier storage) can be resolved by switching from using `emptyDir` like below
```
      volumes:
        - name: mem
          emptyDir:
            medium: "Memory"
            sizeLimit: 1G
```
to a PVC.
```
      volumes:
        - name: mem
          persistentVolumeClaim:
            claimName: mem-pvc
```
The trick is, although there's only ONE PVC-PV pair for ALL worker daemon pods, this PV can be a `hostPath` or `local` type. The effect will be each worker pod reads and writes to its local path. This has been manually verified to work (both `hostPath` and `local`).

The drawback is, if this PV is NOT LOCAL (like AWS EBS), ALL worker pods will point to this one EBS volume, which will not work. @madanadit Do you think workers should also consider using non-local PVs for tiered storage?

**Comment by madanadit** [comment]
@jiacheliu3 we should start with `local` PV to use local SSDs launched for example in gke as https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/local-ssd

**Comment by jiacheliu3** [comment]
After some research, I think the best practice for using `local` PV is by declaring a PVC on the Pod side. Then there are 2 ways to satisfy the PVC:
1. You can manually create local PV. 

2. Use a provisioner to create local PVs for you like:
https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/docs/getting-started.md
This is the static provisioner that K8s doc refers to.

In short, to use `local` PV, we need to support PVC for the declaration, and provide appropriate documentation for the user to provision the PVs for the PVCs.

**Comment by jiacheliu3** [comment]
3) and 4) are resolved by #11175 and #11171. Now this has been fully addressed and can be closed.
